### ISSUE TITLE
404 error on version 1.9.10

### ISSUE THREAD

**Issue by mymikemiller** [Hexer10/youtube_explode_dart#144]
I'm getting the same error as described in [this issue](https://github.com/Hexer10/youtube_explode_dart/issues/132) that was marked fixed a month ago in version 1.9.5, but I'm experiencing the issue in version 1.9.10 no matter what videoId I use. The error is as follows:

```
FatalFailureException (FatalFailureException: Failed to perform an HTTP request to YouTube due to a fatal failure.
In most cases, this error indicates that YouTube most likely changed something, which broke the library.
If this issue persists, please report it on the project's GitHub page.
Request: GET https://youtube.com/get_video_info?video_id=uG4h5dp0XsI&el=embedded&eurl=https%3A%2F%2Fyoutube.googleapis.com%2Fv%2FuG4h5dp0XsI&hl=en&html5=1&c=TVHTML5&cver=6.20180913
Response: (404)
```

An example of the call I'm trying to make:

```
var yt = YoutubeExplode();
var video = await yt.videos.get('uG4h5dp0XsI');
```

**Comment by Cdriic** [comment]
> I'm getting the same error as described in [this issue](https://github.com/Hexer10/youtube_explode_dart/issues/132) that was marked fixed a month ago in version 1.9.5, but I'm experiencing the issue in version 1.9.10 no matter what videoId I use. The error is as follows:
> 
> ```
> FatalFailureException (FatalFailureException: Failed to perform an HTTP request to YouTube due to a fatal failure.
> In most cases, this error indicates that YouTube most likely changed something, which broke the library.
> If this issue persists, please report it on the project's GitHub page.
> Request: GET https://youtube.com/get_video_info?video_id=uG4h5dp0XsI&el=embedded&eurl=https%3A%2F%2Fyoutube.googleapis.com%2Fv%2FuG4h5dp0XsI&hl=en&html5=1&c=TVHTML5&cver=6.20180913
> Response: (404)
> ```
> 
> An example of the call I'm trying to make:
> 
> ```
> var yt = YoutubeExplode();
> var video = await yt.videos.get('uG4h5dp0XsI');
> ```

same issue, since yesterday

**Comment by Hexer10** [comment]
Seems youtube removed (again) the `get_video_info`. I've updated the way the video info is extracted in v`1.10.0` tough it requires dart `2.13.0` now, since the new release is part of some other changes that were already going on.

Note that it won't be possibile to extract age-restricted videos at all now

**Comment by lrorpilla** [comment]
That's quite a sizeable amount of videos now inaccessible. May I know the cause of the limitation preventing age-restricted videos from being extracted? Is this a temporary sticking point or is this change something that can't be worked around?

**Comment by Hexer10** [comment]
@lrorpilla Before YouTube patched this out, there was a way to extract age-limited videos(which usually require to be logged in) using some sort of workarounds. But now there is no way to access those videos anymore, the only solution would be to pass the cookies of a logged in user

**Comment by 89z** [comment]
https://github.com/Hexer10/youtube_explode_dart/issues/147

**Comment by mymikemiller** [comment]
Thing are working for me now, thanks for the quick fix!

I used to work for a company that did web scraping of bank websites and we had to do all kinds of things to get around the anti-bot measures those banks put in place, often powered by Akamai. It was definitely a last resort since it required a lot more processing power for the bulk scraping we were doing, but when nothing else worked we could usually outsmart the bots by using [Puppeteer](https://developers.google.com/web/tools/puppeteer/). Not sure if it would be worth looking into that as a way to get around the age-limited video block. It might, at least, provide an easy way to get a hold of a logged in user's cookies for the rest of the requests.
